posts/2023-01-20-types-announcement.md

```diff
@@ -42,9 +42,9 @@ As mentioned above, a nearly essential element of the growing Rust language is t
 
 As far back as 2015, not long after the release of Rust 1.0, an experimental Rust trait solver called Chalk began to be written. The core idea of Chalk is to translate the surface syntax and ideas of the Rust trait system (e.g. traits, impls, where clauses) into a set of logic rules that can be solved using a Prolog-like solver. Then, once this set of logic and solving reaches parity with the trait solver within the compiler itself, the plan was to simply replace the existing solver. In the meantime (and continuing forward), this new solver could be used by other tools, such as rust-analyzer, where it is used today.
 
-Now, given Chalk's age and the promises it had been hoped to be able to deliver on, you might be tempted to ask the question "Chalk, when?" - and plenty have. However, we've learned over the years that Chalk is likely not the correct long-term solution for Rust, for a few reasons. First, as mentioned a few times in this post, the trait solver is only but a part of a larger type system; and modeling how the entire type system fits together gives a more complete picture of its details than trying to model the parts separately. Second, the needs of the *compiler* are quite different than the needs of a *formalization*: the compiler needs performant code with the ability to track information required for powerful diagnostics; a good formalization is one that is not only complete, but also easy to maintain, read, and understand. Over the years, Chalk has tried to have both and it has so far ended up with neither.
+Now, given Chalk's age and the promises it had hoped to be able to deliver on, you might be tempted to ask the question "Chalk, when?" - and plenty have. However, we've learned over the years that Chalk is likely not the correct long-term solution for Rust, for a few reasons. First, as mentioned a few times in this post, the trait solver is only but a part of a larger type system; and modeling how the entire type system fits together gives a more complete picture of its details than trying to model the parts separately. Second, the needs of the *compiler* are quite different than the needs of a *formalization*: the compiler needs performant code with the ability to track information required for powerful diagnostics; a good formalization is one that is not only complete, but also easy to maintain, read, and understand. Over the years, Chalk has tried to have both and it has so far ended up with neither.
```

Comment:
I'm not sure this is correct. "it had been hoped" means "some anonymous group hoped". "it had hoped" means Chalk itself is doing the hoping, which seems odd.

Reply:
Oh, I see. In the case that it's some group who's hoping, it should probably be "...and the promises it had been hoped _it would_ be able to deliver on" instead. But that feels clunky. If I was doing the editing I might re-write it as "...and the promises [group] had hoped it would be able to deliver on..." Actually naming the group, even if it's just something like "many had hoped..." But at this point I don't know who this was intended to refer to.

Reply:
So, realizing that this PR never got merged.

Ultimately, I'm happy to keep it `it had been hoped to be able to deliver on" or "it had been hoped it would be able to deliver on".

Once you change that, I should be able to merge this PR.